db/db_impl/db_impl.h

```diff
@@ -1559,22 +1558,38 @@ class DBImpl : public DB {
   // Assign db_id_ and write DB ID to manifest if necessary.
   void SetDBId(std::string&& id, bool read_only, RecoveryContext* recovery_ctx);
 
+  // Collect a deduplicated collection of paths used by this DB, including
+  // dbname_, DBOptions.db_paths, ColumnFamilyOptions.cf_paths. These paths
+  // cannot be dynamically changed, so they are collected once during DB
```

Comment:
Is this true even if `CreateColumnFamily` is called?

Reply:
That's a great point. This is indeed not true when new column families are added during DB session, so this set needs to be refreshed there too. Thank you for catching this!

Reply:
I totally missed how column family operations would affect this set. The whole assumption that this set is fixed once DB initialized and is kept intact through the whole DB session is not true at all. So it's not useful to have a member variable storing them like this. Although I can refresh the member variable when column families are updated, the case for when column families are added are relatively straightforward but the case for when a column family becomes completely dereferenced is more nuanced. 

So instead I just make the function a pure util function for collecting these paths on demand, it's just a code dedupe refactor.